spark/src/main/java/org/apache/iceberg/spark/source/Writer.java

```diff
@@ -114,6 +115,7 @@
     long tableTargetFileSize = PropertyUtil.propertyAsLong(
         table.properties(), WRITE_TARGET_FILE_SIZE_BYTES, WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT);
     this.targetFileSize = options.getLong("target-file-size-bytes", tableTargetFileSize);
+    this.useWriterLengthAsFileSize = options.getBoolean("use-writer-length-as-file-size", false);
```

Comment:
This shouldn't be a new option. Let's remove it.

Reply:
@rdblue  do you suggestion always using writer length, instead of calling `dataFileBuilder.withEncryptedOutputFile(currentFile)` which internally calls getStatus for s3a?

Reply:
Yes, I think it should always use the length reported by the writer.

Reply:
sounds good I will update PR.

Reply:
@rdblue please take a look at updated PR at your convenience.